src/modules/facility/facility.controller.ts

```diff
@@ -0,0 +1,66 @@
+import { Controller, Get, Param } from '@nestjs/common';
+import { ApiInternalServerErrorResponse, ApiNotFoundResponse, ApiOkResponse, ApiOperation, ApiParam } from '@nestjs/swagger';
+
+import { GetFacilityByIdentifierResponseDto } from './dto/get-facility-by-identifier-response.dto';
+import { FacilityService } from './facility.service';
+
+@Controller('facilities')
+export class FacilityController {
+  constructor(private readonly facilityService: FacilityService) {}
+
+  @Get(':facilityIdentifier')
+  @ApiOperation({ summary: 'Get the facility matching the specified facility identifier.' })
+  @ApiParam({
+    name: 'facilityIdentifier',
+    required: true,
+    type: 'string',
+    description: 'The identifier of the facility in ACBS.',
+    example: '0000000001',
+  })
+  @ApiOkResponse({
+    description: 'The facility has been successfully retrieved.',
+    type: GetFacilityByIdentifierResponseDto,
+  })
+  @ApiNotFoundResponse({
+    description: 'The specified facility was not found.',
+  })
+  @ApiInternalServerErrorResponse({
+    description: 'An internal server error has occurred.',
+  })
+  async getFacilityByIdentifier(@Param('facilityIdentifier') facilityIdentifier: string): Promise<GetFacilityByIdentifierResponseDto> {
```

Comment:
Is this input validated?
We should allow just 10 digit strings. Other input should return 400.

Reply:
@avaitonis I think we should drop this requirement to validate the partyId / facilityId / dealId in the URLs because ACBS should tell us that the party/facility/deal doesn't exist anyway so we'll return a 404. I think that's good enough instead of making extra development + testing work for ourselves. What do you think?

Reply:
Good spot, I've added validation. I think we should also create reusable tests that check that we return a 400 when we use invalid URL params (similar to the withStringFieldValidationApiTests we have for request body fields), but I've raised a separate ticket for this: https://ukef-dtfs.atlassian.net/browse/APIM-202

Reply:
@lmarrai-sw @avaitonis let's discuss this in the PR meeting later.

Reply:
In light of call, we will validate URL params and return a 400. Tests can be written in a separate ticket.